tests/aliquot/test_aliquot_models.py

```diff
@@ -0,0 +1,383 @@
+        s1 = Sample.query.filter_by(external_id=sample_id).all()
+        s2 = Sample.query.filter_by(external_id=s.external_id).all()
+        
+        a1 = Aliquot.query.filter_by(external_id=aliquot_id).all()
+        a2 = Aliquot.query.filter_by(external_id=a.external_id).all()
+        
+        #update a1 and a2
+        a1[0].analyte_type = 'RNA'
+        a2[0].volume = 13.89
+        
+        #get updated aliquots
+        a1 = Aliquot.query.filter_by(external_id=aliquot_id).all()
+        a2 = Aliquot.query.filter_by(external_id=a.external_id).all()
+        self.assertEqual(Participant.query.count(), 1)
+        self.assertEqual(p[0].samples[0].external_id,'Test_Sample_0')
+        self.assertEqual(p[0].samples[1].external_id,'Test_Sample_1')
+
+        self.assertEqual(Sample.query.count(), 2)
+        self.assertEqual(s1[0].aliquots[0].external_id,'Test_Aliquot_0')
+        self.assertEqual(s2[0].aliquots[0].external_id,'Test_Aliquot_1')
+
+        self.assertEqual(Aliquot.query.count(), 2)
+        self.assertEqual(a1[0].analyte_type,'RNA')
+        self.assertEqual(a2[0].volume,13.89)
+
+    def test_one_to_many_relationship_delete(self):
+        """
+        Test deleting one of the aliquot in samples
+        """
+        dt = datetime.now()
+        #create a participant with a sample and a aliquot
+        participant_id, sample_id, aliquot_id= self.create_participant_sample_aliquot()
+        p = Participant.query.filter_by(external_id=participant_id).one_or_none()
+        
+        #adding another sample to participant
+        s = Sample(external_id='Test_Sample_1', tissue_type='Normal', composition='Test_comp_1', 
+                   anatomical_site='Brain', age_at_event_days=456, tumor_descriptor='Metastatic', participant_id =p.kf_id)
+        
+        db.session.add(s)
+        db.session.commit()
+
+        #adding another aliquot to sample
+        aliquot_data={
+        'external_id':'Test_Aliquot_1',
+        'shipment_origin':'CORIELL',
+        'shipment_destination':'Broad Institute',
+        'analyte_type':'DNA',
+        'concentration':100,
+        'volume':12.67,
+        'shipment_date':dt
+        }
+        a = Aliquot(**aliquot_data, sample_id=s.kf_id)
+        db.session.add(a)
+        db.session.commit()
+
+        #p-s1-a1
+        #p-s2-a2
+        # Get participant, samples and aliquots
+
+        p = Participant.query.filter_by(external_id=participant_id).all()
+        
+        s1 = Sample.query.filter_by(external_id=sample_id).one_or_none()
+        s2 = Sample.query.filter_by(external_id=s.external_id).one_or_none()
+        
+        a1 = Aliquot.query.filter_by(external_id=aliquot_id).one_or_none()
+        a2 = Aliquot.query.filter_by(external_id=a.external_id).one_or_none()
+        
+        #delete a1
+        db.session.delete(a1)
+        db.session.commit()
+        
+        self.assertEqual(Participant.query.count(), 1)
+        self.assertEqual(Sample.query.count(), 1)
```

Comment:
Hmm so deleting the aliquot will also delete the its parent sample?

Reply:
its because I have one aliquot for the sample. This is the example I tested ;a participant has two samples , each sample has one aliquot.

Reply:
right, but is that sample is still connected to a participant? It seems that even though that sample no longer has any aliquots, it's still not an orphan, so it doesn't make sense to me for it to be deleted.

Reply:
I think aliquot ID is representative of the sample. so if we don't have an aliquot ID, they woudn't be any sample.aliquot is like the base unit (so to speak) of a sample.This is what Elizabeth told me. @allisonheath does a sample exists with out an aliquot?

Reply:
Yes a sample can exist without an aliquot. In reality, a sample always exists and an aliquot is either a piece of that sample or a processed piece of that sample. So I don't think they should be tightly coupled like this in general. In our case, the "sample" here is the top-level aliquot and the "aliquot" is the aliquot that's actually shipped to the sequencing center. We track a lot more "aliquots" in the intermediary, but that's doesn't generalize to other biospecimen processes. 

One issue for KF from most cohorts we often only have the aliquot information, nothing about the original sample taken from the participant. So there's a question of do we want the aliquot to go back directly to the participant as well in this case? Or create an artificial sample because it should have existed?

Reply:
To make myself clear , these are the cases right ?
1)If there is no sample information, but there is aliquot information then sample entity is just like a pointer that connects to participant
2)If Sample exists, no aliquots are extracted from it. Sample entity exists ;if any aliquot is extracted from this sample; those aliquot_ids are attached to this sample ; For now, in either case sample exists.